internal/orchestrion/context_stack.go

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package orchestrion
+
+// contextStack is the object put in the GLS slot of runtime.g inserted by orchestrion. it is used to store context values
+// that are shared across the same goroutine.
+// TODO: handle cross-goroutine context values
+type contextStack map[any][]any
+
+// getDDContextStack is a main way to access the GLS slot of runtime.g inserted by orchestrion. This function must not be
+// called if the enabled variable is false.
+func getDDContextStack() *contextStack {
```

Comment:
>This function must not be called if the enabled variable is false.

Why? What happens if you did?

Reply:
You would get a new contextStack that will be garbage-collected 10 instructions later :shrug: . "Must" is a little strong here but anyway this is unexported and the doc on this is already free of charges